### PR TITLE
chore: various vscode workspace related fixes

### PR DESCRIPTION
**Pull request by braddialpad**
## Description
Fixed issues with the vscode workspace configuration

- Wasn't running eslint on dot folders (.vuepress) since changing to the monorepo.
- Added some additional useful options that should be set by default.
- Configured vitest workspace, this is required for the vitest vscode extension to run on monorepo.
- Added extension recommmendations, should assist with onboarding.
- We had two separate markdownlint configurations. Consolidated to one.
- vitest updated to 1.0! (just released last week)

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: dialtone.code-workspace [hunk 1]
```diff
@@ -1,73 +1,87 @@
...
     },
 
-		// SEARCH
+    // SEARCH
     // Note that gitignored files are automatically excluded from search. Do not add files
     // here that are already gitignored.
     "search.exclude": {
-      "**/package-lock.json": true,
-      "**/locale": true
+      "**/package-lock.json": true
```
Comment (juliodialpad) [inline dialtone.code-workspace]: package-lock.json files shouldn't longer exists on the monorepo as we're using pnpm

File: packages/dialtone-vue2/package.json [hunk 2]
```diff
@@ -92,7 +92,7 @@
     "typedoc-plugin-markdown": "^3.16.0",
     "vite": "^5.0.0",
     "vite-plugin-vue2-svg": "^0.4.0",
-    "vitest": "^0.34.5",
+    "vitest": "^1.0.4",
```
Comment (juliodialpad) [inline packages/dialtone-vue2/package.json]: You can remove the dependency here if it's already on root's package.json

File: packages/dialtone-vue3/package.json [hunk 3]
```diff
@@ -91,7 +91,7 @@
     "typedoc-plugin-markdown": "^3.16.0",
     "vite": "^5.0.0",
     "vite-svg-loader": "^4.0.0",
-    "vitest": "0.33.0",
+    "vitest": "1.0.4",
```
Comment (juliodialpad) [inline packages/dialtone-vue3/package.json]: Same here

File: package.json [hunk 4]
```diff
@@ -11,7 +11,8 @@
     "start:dialtone-vue3": "nx run-many --target=start --projects=dialtone,dialtone-vue3",
     "start:dialtone-vue2": "nx run-many --target=start --projects=dialtone,dialtone-vue2",
     "prepare": "husky install",
-    "release": "./scripts/release.sh"
+    "release": "./scripts/release.sh",
+    "test:vitest": "vitest"
```
Comment (juliodialpad) [inline package.json:15]: Nice, it's going to be easier to maintain just one instance of vitest

**Review by juliodialpad — CHANGES_REQUESTED** [review]
Nice work! Couple changes

File: dialtone.code-workspace (hunk 1 above)
Comment (braddialpad) [reply dialtone.code-workspace]: right, I will ignore the pnpm lock instead

File: packages/dialtone-vue2/package.json (hunk 2 above)
Comment (braddialpad) [reply packages/dialtone-vue2/package.json]: Yeah I was wondering how that worked... do all sub packages inherit the parent packages?

**Comment by braddialpad** [comment]
Fixes are in

**Comment by iropolo** [comment]
Would we cool to have @ninamarina review here since she is using vscode and this would be helpful.

**Review by iropolo — APPROVED** [review]

File: packages/dialtone-vue2/package.json (hunk 2 above)
Comment (juliodialpad) [reply packages/dialtone-vue2/package.json]: I think it's not inheriting, as you're running `vitest` from the root package, the other packages don't really need to have that dependency.

**Pull request merged**